internal/resources/policy/kind/network/recipe/allow_all_flatten_test.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package recipe
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	policyrecipenetworkmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/network"
+)
+
+func TestFlattenAllowAll(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		description string
+		input       *policyrecipenetworkmodel.V1alpha1CommonPolicySpecNetworkV1AllowAll
+		expected    []interface{}
+	}{
+		{
+			description: "check for nil allow-all recipe network policy",
+			input:       nil,
+			expected:    nil,
+		},
+		{
+			description: "normal scenario with valid allow-all recipe network policy",
```

Comment:
you are missing test case to test :

if allowAll.FromOwnNamespace == nil {
return data
}

Reply:
Added the case